calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -505,3 +518,103 @@ func (rh resourceHelper) GetTableTemplate(headings []string) (string, error) {
 
 	return buf.String(), nil
 }
+
+// validateMetadata compares the Metadata for a ResourceObject and warns if anything was
+// changed that will not be changed as a part of the update. Any discrepancies to the metadata
+// will be changed to match the metadata available in the stored object except for Labels and
+// Annotations.
+func validateMetadata(stored, changed ResourceObject) ResourceObject {
+	sm := stored.GetObjectMeta()
+	cm := changed.GetObjectMeta()
+
+	// Set the fields that are allowed to be overwritten (Labels and Annotations)
+	// so that they will not be overwritten.
+	// TODO: Figure out if a warning is needed
+	sm.SetAnnotations(cm.GetAnnotations())
+	sm.SetLabels(cm.GetLabels())
+
+	// Make a copy of the resource so we do not need to modify the input
+	resOut := changed.DeepCopyObject().(ResourceObject)
```

Comment:
Does it matter if we modify the input in this case, I don't think anything else will need the unprocessed "changed" resource.

Reply:
or is this thinking ahead to the editor or something?

Reply:
I did not have a specific feature in mind when doing this (though the editor may definitely run into this at a future date). It shouldn't make a functional difference now, just thought it would keep things cleaner if we didn't have to worry about the side effects of copying to and from the metadata.  I can remove it, but I think it might be nice to leave the return value so its obvious which resource is the one to use in the update step.